.changeset/fuzzy-lizards-end.md

```diff
@@ -0,0 +1,34 @@
+---
+'@solana/react': minor
+---
+
+Added a `useSignIn` hook that, given a `UiWallet` or `UiWalletAccount`, returns a function that you can call to trigger a wallet's [&lsquo;Sign In With Solana&rsquo;](https://phantom.app/learn/developers/sign-in-with-solana) feature.
+
+#### Example
+
+```tsx
+import { useSignIn } from '@solana/react';
+
+function SignInButton({ wallet }) {
+    const csrfToken = useCsrfToken();
+    const signIn = useSignIn(wallet);
+    return (
+        <button
+            onClick={async () => {
+                try {
+                    const { account, signedMessage, signature } = await signIn({
+                        domain: window.location.host,
```

Comment:
`domain` probably not needed in the example since apps shouldn't provide it.

Reply:
You're right about that! I noticed some wallets fail awkwardly when you don't supply it, but that's a bug with their implementation.

Reply:
For posterity, one such error when you don't supply `domain`.

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/lE1zD7Jpg6mWNv3djNKZ/42e1e45f-f04f-4edf-9f92-e3a424d9bfb5.png)